src/main/java/uk/ac/ebi/ega/egafuse/service/EgaFile.java

```diff
@@ -17,32 +17,95 @@
  */
 package uk.ac.ebi.ega.egafuse.service;
 
+import static uk.ac.ebi.ega.egafuse.config.EgaFuseApplicationConfig.NUM_PAGES;
+import static uk.ac.ebi.ega.egafuse.config.EgaFuseApplicationConfig.PAGE_SIZE;
+
+import java.io.IOException;
+import java.util.HashSet;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import jnr.ffi.Pointer;
 import ru.serce.jnrfuse.struct.FileStat;
 import uk.ac.ebi.ega.egafuse.model.File;
 
 public class EgaFile extends EgaPath {
+    private static final Logger LOGGER = LoggerFactory.getLogger(EgaFile.class);
     private File file;
-    private String type;
+    private EgaFileService egaFileService;
+    private HashSet<Integer> keys = new HashSet<>();
 
     public EgaFile(String name, EgaDirectory parent) {
         super(name, parent);
     }
 
-    public EgaFile(String name, String type, File file) {
+    public EgaFile(String name, File file, EgaFileService egaFileService) {
         super(name);
         this.file = file;
-        this.type = type;
+        this.egaFileService = egaFileService;
     }
-    
+
     @Override
     public void getattr(FileStat stat) {
         stat.st_mode.set(FileStat.S_IFREG | 0444);
-        stat.st_size.set(type.equalsIgnoreCase("CIP") ? file.getFileSize() - 16 : file.getFileSize());
+        stat.st_size.set(file.getFileSize());
     }
 
     public int read(Pointer buffer, long size, long offset) {
-        return 0;
+        long fsize = file.getFileSize();
```

Comment:
All of this should not be in the logic of the File, he does not know and does not need to know about caches, that is something up to the service, encapsulate it there. The telltale sign is that you are using constants defined in another class that should be private. And bringing concepts about pages which is something that this class should not understand, this class should only know about bytes, how many and offset.

Reply:
Done